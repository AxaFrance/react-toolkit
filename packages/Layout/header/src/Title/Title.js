import React from 'react';
import PropTypes from 'prop-types';
import { ClassManager, Constants } from '@axa-fr/react-toolkit-core';
import Action from '@axa-fr/react-toolkit-action';
import ToggleButton from '../ToggleButton';

const propTypes = {
  ...Constants.propTypes,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  children: PropTypes.node,
  toggleMenu: PropTypes.func,
};

const defaultClassName = 'af-title-bar';
const defaultProps = {
  ...Constants.defaultProps,
  className: defaultClassName,
  children: null,
  subtitle: '',
  toggleMenu: null,
};

const Title = props => {
  const {
    title,
    subtitle,
    children,
    toggleMenu,
    className,
    classModifier,
  } = props;
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      <div className={`container ${defaultClassName}__wrapper`}>
        <ToggleButton idControl="mainmenu">
          <Action
            className="btn af-btn--circle af-title-bar__mobile-menu"
            classModifier={null}
            id="togglemenu"
            icon="menu-hamburger"
            title="Toggle menu"
            onClick={toggleMenu}
          />
        </ToggleButton>
        <h1 className={`${defaultClassName}__title`}>
          {title}
          {subtitle && (
            <small className={`${defaultClassName}__subtitle`}>
              {subtitle}
            </small>
          )}
        </h1>
        {children}
      </div>
    </div>
  );
};

Title.propTypes = propTypes;
Title.defaultProps = defaultProps;

export default Title;
