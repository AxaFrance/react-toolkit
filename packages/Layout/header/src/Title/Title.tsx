import React, { ReactNode } from 'react';
import { ClassManager } from '@axa-fr/react-toolkit-core';
import Action from '@axa-fr/react-toolkit-action';
import ToggleButton from '../ToggleButton';

const defaultClassName = 'af-title-bar';

type Props = {
  title: string;
  subtitle?: string;
  children?: ReactNode;
  toggleMenu?: () => void;
  className?: string;
  classModifier?: string;
};
const Title = ({
  title,
  subtitle,
  children,
  toggleMenu,
  className,
  classModifier,
}: Props) => {
  const componentClassName = ClassManager.getComponentClassName(
    className,
    classModifier,
    defaultClassName
  );
  return (
    <div className={componentClassName}>
      <div className={`container ${defaultClassName}__wrapper`}>
        {!!toggleMenu && (
          <div className="burger-container">
            <ToggleButton idControl="mainmenu">
              <Action
                className="btn af-title-bar__mobile-menu af-btn--circle"
                id="togglemenu"
                icon="menu-hamburger"
                title="Toggle menu"
                onClick={toggleMenu}
              />
            </ToggleButton>
          </div>
        )}
        <h1 className={`${defaultClassName}__title`}>
          {title}
          {subtitle && (
            <small className={`${defaultClassName}__subtitle`}>
              {subtitle}
            </small>
          )}
        </h1>
        {children}
      </div>
    </div>
  );
};

export default Title;
