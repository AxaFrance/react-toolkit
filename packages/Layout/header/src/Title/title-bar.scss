@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.af-title-bar {
  background: $brand-primary;
  color: $white;
  overflow: auto;
  padding: 0.66rem 0;
  margin-bottom: 2rem;

  &--fixed {
    position: fixed;
    width: 100%;
  }

  &__button {
    margin-right: 1rem;

    .button--circled:hover {
      color: initial;

      .button__icon {
        fill: initial;
      }
    }
  }

  &__title {
    display: inline-block;
    margin: 0;
    color: $white;
    font-weight: normal;
  }

  &__subtitle {
    margin-left: 0.25rem;
    color: $color-tarif-table-header-selected7;
  }

  &__actions {
    .af-btn--circle {
      vertical-align: middle;
      margin-left: 1rem;

      &:focus {
        box-shadow: 0 0 6px $white;
        color: $white;
      }
    }
  }

  &__link {
    color: $white;
    text-decoration: underline;

    &:hover {
      opacity: 0.65;
      color: $white;
    }

    &:focus {
      text-shadow: 0 0 5px $white;
      color: $white;
    }
  }

  &__mobile-menu {
    display: inline-block;
    background-image: none;
    background-color: transparent;
    padding: 0;
    margin-right: 1rem;
    .glyphicon {
      fill: $white;
      width: 17px;
    }
  }

  &__icon {
    font-size: 1em;
  }
}

@include media-breakpoint-up(sm) {
  .af-title-bar {
    padding: 0.7em 0;

    &__wrapper {
      display: flex;
      align-items: center;
      flex-wrap: wrap;
      justify-content: space-between;
      width: 100%;
    }
  }

  .burger-container {
    display: none;
  }
}
@include media-breakpoint-up(md) {
  .af-title-bar {
    padding: 0.66rem 0;

    &__wrapper {
      width: nth(nth($container-max-widths, 4), 2);
      max-width: 100%;
    }

    &__title {
      font-size: 1.25rem;
    }
  }
}
