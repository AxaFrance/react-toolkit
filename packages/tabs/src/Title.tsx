import React from 'react';
import {
  withClassDefault,
  withClickId,
  WithClassModifierOptions,
  withClassModifier,
  compose,
  withProps,
} from '@axa-fr/react-toolkit-core';

const onChangeEvent = 'onChange';

export interface TitleComponentProps {
  enable?: boolean | null;
  active: boolean;
  className?: string;
  classModifier?: string;
  id?: string;
}

export interface TitleHandlerProps {
  onChange: React.MouseEventHandler<HTMLButtonElement>;
}

const Title: React.SFC<TitleComponentProps & TitleHandlerProps> = ({
  className,
  onChange,
  children,
  id,
}) => (
  <li className={className}>
    {/* eslint-disable-next-line react/button-has-type */}
    <button id={id} className="af-tabs__link" onClick={onChange}>
      {children}
    </button>
  </li>
);

const DEFAULT_CLASSNAME = 'af-tabs__item';

export type TitleProps = TitleComponentProps &
  TitleHandlerProps &
  WithClassModifierOptions;

const setWithProps = (props: TitleComponentProps) => ({
  ...props,
  classModifier: (props.classModifier || '').concat(
    (props.enable === false ? ' disabled' : '').concat(
      props.active ? ' active' : ''
    )
  ),
});

const enchance = compose<TitleComponentProps & TitleHandlerProps, TitleProps>(
  withProps<TitleComponentProps, TitleProps>(setWithProps),
  withClickId({ event: [onChangeEvent] }),
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
)(Title);

enchance.displayName = 'Title';

export default enchance;
