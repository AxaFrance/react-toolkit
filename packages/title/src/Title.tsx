import React from 'react';
import {
  PropsManager,
  withClassDefault,
  withClassModifier,
  WithClassModifierOptions,
  compose,
} from '@axa-fr/react-toolkit-core';

const DEFAULT_CLASSNAME = 'af-title';

interface TitleBaseProps extends React.HTMLProps<HTMLHeadingElement> {}

const defaultProps: Partial<TitleBaseProps> = {
  children: null,
};

const omitProperties = PropsManager.omit(['classModifier']);

const TitleRaw: React.FC<TitleBaseProps> = ({ children, ...otherProps }) => (
  <h1 {...omitProperties(otherProps)}>{children}</h1>
);

TitleRaw.defaultProps = defaultProps;

export interface TitleProps extends WithClassModifierOptions, TitleBaseProps {}

const enhance = compose<TitleBaseProps, TitleProps>(
  withClassDefault(DEFAULT_CLASSNAME),
  withClassModifier
);

const Enhanced = enhance(TitleRaw);
Enhanced.displayName = 'Title';
export default Enhanced;
