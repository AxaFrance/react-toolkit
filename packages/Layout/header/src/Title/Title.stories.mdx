import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { ArgsTable } from '@storybook/addon-docs';
import { action } from '@storybook/addon-actions';
import Action from '@axa-fr/react-toolkit-action';
import Title from './Title';

<Meta title="Structure/Title Bar" component={Title} argTypes={{}} />

# Title

export const Template = ({ children, ...args }) => (
  <Title {...args}>{children}</Title>
);

The header package exports an `Title` in order to display a title bar.  
To do so, you need to specify the `title` property. If you want more information, there is a `subtitle` property that allows you do that.  
You can then pass some `children` to display even more information on your title bar, for example for actions buttons on the right side of the bar (by using the `af-title-bar__actions` css class like in this complex story example).  
You can as well display an toggle menu on the left of the title bar. To do so, you need to specify the `toggleMenu` property which is a function that will be called when you will press the toggle button shown on the page.

Don't forget to load the css file of the component.

```javascript
import '@axa-fr/react-toolkit-layout-header/dist/Title/af-title-bar.css';
```

<Canvas>
  <Story
    name="Default"
    args={{
      title: 'Toolkit Axa',
      subtitle: 'Info complémentaire',
    }}>
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Default" />

## With actions buttons as children

export const actions = [
  {
    id: '001',
    href: '#enregistrer',
    icon: 'floppy-disk',
    title: 'Enregistrer',
  },
  {
    id: '002',
    href: '#imprimer',
    icon: 'print',
    title: 'Imprimer',
  },
];

<Canvas>
  <Story
    name="With actions"
    args={{
      title: 'Toolkit Axa',
      subtitle: 'Info complémentaire',
      children: (
        <div className="af-title-bar__actions">
          <a className="af-title-bar__link" href="#lien" title="lien titlebar">
            lien titlebar
          </a>
          {actions.map(({ icon, href, title, id }) => (
            <Action
              key={id}
              icon={icon}
              href={href}
              title={title}
              onClick={() => {
                action('Action click');
              }}
            />
          ))}
        </div>
      ),
    }}>
    {Template.bind({})}
  </Story>
</Canvas>
