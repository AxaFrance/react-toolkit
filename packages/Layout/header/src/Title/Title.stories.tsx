import React from 'react';
import { action } from '@storybook/addon-actions';
import { Meta } from '@storybook/react';
import Action from '@axa-fr/react-toolkit-action';
import Title from './Title';
import readme from '../../README.md';

export default {
  title: 'Layout/Header/Title',
  component: Title,
  parameters: {
    readme: {
      sidebar: readme,
    },
    options: {},
  },
} as Meta;

const actions = [
  {
    id: '001',
    href: '#enregistrer',
    icon: 'floppy-disk',
    title: 'Enregistrer',
  },
  {
    id: '002',
    href: '#imprimer',
    icon: 'print',
    title: 'Imprimer',
  },
];

export const Default = (args) => <Title {...args} />;
Default.args = {
  title: 'Toolkit Axa',
  subtitle: 'Info complÃ©mentaire',
};
Default.argTypes = {
  toggleMenu: { action: 'onToggle' },
};

export const Complex = (args) => (
  <Title {...args}>
    <div className="af-title-bar__actions">
      <a className="af-title-bar__link" href="#lien" title="lien titlebar">
        lien titlebar
      </a>
      {actions.map(({ icon, href, title, id }) => (
        <Action
          key={id}
          icon={icon}
          href={href}
          title={title}
          onClick={() => {
            action('Action click');
          }}
        />
      ))}
    </div>
  </Title>
);
Complex.args = {
  ...Default.args,
};
Complex.argTypes = {
  ...Default.argTypes,
};
