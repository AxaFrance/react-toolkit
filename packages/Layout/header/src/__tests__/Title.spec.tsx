import React from 'react';
import { render } from '@testing-library/react';
import { Title } from '../Title/index';

describe('<Title>', () => {
  it('renders Title correctly', () => {
    const { asFragment } = render(
      <Title
        title="Toolkit Axa"
        subtitle="Info complémentaire"
        toggleMenu={() => null}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders Title correctly without menu', () => {
    const { asFragment } = render(
      <Title title="Toolkit Axa" subtitle="Info complémentaire" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
