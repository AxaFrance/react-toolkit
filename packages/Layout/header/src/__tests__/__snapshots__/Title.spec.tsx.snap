// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title> renders Title correctly 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <div
        class="burger-container"
      >
        <a
          aria-controls="mainmenu"
          aria-haspopup="true"
          class="btn af-title-bar__mobile-menu af-btn--circle"
          href="#"
          id="togglemenu"
          role="button"
          tabindex="0"
          title="Toggle menu"
        >
          <i
            class="glyphicon glyphicon-menu-hamburger"
          />
        </a>
      </div>
      <h1
        class="af-title-bar__title"
      >
        Toolkit Axa
        <small
          class="af-title-bar__subtitle"
        >
          Info complémentaire
        </small>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Title> renders Title correctly without menu 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar"
  >
    <div
      class="container af-title-bar__wrapper"
    >
      <h1
        class="af-title-bar__title"
      >
        Toolkit Axa
        <small
          class="af-title-bar__subtitle"
        >
          Info complémentaire
        </small>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
